### PR TITLE
Skip user errors from retrying to provision L4 service

### PR DESCRIPTION
**Pull request by cezarygerard**

### CODE REVIEW

**Comment by aojea** [comment]
it makes sense to not retry errors we know we'll not be able to reconcile, I don't know where is the best way to implement it though

File: pkg/l4lb/l4lbcommon.go [hunk 1]
```diff
@@ -111,3 +111,10 @@ func isHealthCheckDeleted(cloud *gce.Cloud, hcName string) bool {
 	_, err := composite.GetHealthCheck(cloud, meta.GlobalKey(hcName), meta.VersionGA)
 	return utils.IsNotFoundError(err)
 }
+
+func skipUserError(err error) error {
```
Comment (mmamczur) [inline pkg/l4lb/l4lbcommon.go:115]: if we skip, will this still be emitted as event for the service? so that the users get the error message

**Comment by code-elinka** [comment]
> it makes sense to not retry errors we know we'll not be able to reconcile, I don't know where is the best way to implement it though

@aojea Maybe we can use `t.queue.Forget(key)` for it

**Comment by cezarygerard** [comment]
L4 controller has accces to the following queue interface
```
type TaskQueue interface {
	Run()
	Enqueue(objs ...interface{})
	Shutdown()
	Len() int
	NumRequeues(obj interface{}) int
}
```

there is no access to underlying queue
image

File: pkg/l4lb/l4lbcommon.go (hunk 1 above)
Comment (cezarygerard) [reply pkg/l4lb/l4lbcommon.go:115]: yes, we emit the error imperatively

**Pull request merged**